aeron-archive/src/main/java/io/aeron/archive/ReplicationSession.java

```diff
@@ -346,27 +349,38 @@ else if (isDestinationRecordingEmpty)
         }
 
         State nextState = State.EXTEND;
+        String reason = "onRecordingDescriptor";
 
         if (null != liveDestination)
         {
             if (NULL_POSITION != stopPosition)
             {
-                state(State.DONE);
-                error("cannot live merge without active source recording", ArchiveException.GENERIC);
+                final String errorMsg = "cannot live merge without active source recording";
+                state(State.DONE, errorMsg);
+                error(errorMsg, ArchiveException.GENERIC);
                 return;
             }
 
             nextState = State.SRC_RECORDING_POSITION;
+            reason = "has liveDestination";
         }
 
         if (startPosition == stopPosition ||
             (NULL_VALUE != dstRecordingId && stopPosition == catalog.stopPosition(dstRecordingId)))
         {
             signal(stopPosition, SYNC);
             nextState = State.DONE;
+            if (startPosition == stopPosition)
+            {
+                reason = "start/stop positions are the same";
+            }
+            else
+            {
+                reason = "stop position equals catalog stop position";
+            }
```

Comment:
Since we issue a `SYNC` signal here we can simply use "sync" or "in sync" (or maybe "complete") as a reason. I do not see a value in printing how the sync was reached.